packages/vm/lib/runTx.ts

```diff
@@ -136,12 +147,9 @@ export default async function runTx(this: VM, opts: RunTxOpts): Promise<RunTxRes
 
   await state.checkpoint()
 
-  // Is it an Access List transaction?
-  if (
-    'transactionType' in opts.tx &&
-    opts.tx.transactionType === 1 &&
-    this._common.isActivatedEIP(2929)
-  ) {
+  // Typed transaction specific setup tasks
+  if (opts.tx.transactionType !== 0 && this._common.isActivatedEIP(2718)) {
```

Comment:
Ok, removing the EIP-2929 check was switching the error message - which I think is not good - so I added an EIP-2718 check instead. Also refined the surrounding comments a bit so that it gets more encouraged to use this whole section generally for type specific checks in the future and not only the AL things. This should fit better with the initial check on the type + the EIP check.